src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -270,16 +270,134 @@ export class BundleDataClient {
-      expiredDepositsToRefundV3: convertTypedStringRecordIntoNumericRecord(data.expiredDepositsToRefundV3),
-      bundleDepositsV3: convertTypedStringRecordIntoNumericRecord(data.bundleDepositsV3),
-      unexecutableSlowFills: convertTypedStringRecordIntoNumericRecord(data.unexecutableSlowFills),
-      bundleSlowFillsV3: convertTypedStringRecordIntoNumericRecord(data.bundleSlowFillsV3),
+      bundleFillsV3: convertTypedStringRecordIntoNumericRecord(
+        Object.fromEntries(
+          Object.entries(data.bundleFillsV3).map(([chainId, tokenData]) => [
+            chainId,
+            Object.fromEntries(
+              Object.entries(tokenData).map(([token, data]) => [
+                token,
+                {
+                  refunds: data.refunds,
+                  totalRefundAmount: data.totalRefundAmount,
+                  realizedLpFees: data.realizedLpFees,
+                  fills: convertSortableEventFieldsIntoRequiredFields(data.fills),
+                },
+              ])
+            ),
+          ])
+        )
+      ),
+      expiredDepositsToRefundV3: convertTypedStringRecordIntoNumericRecord(
+        Object.fromEntries(
+          Object.entries(data.expiredDepositsToRefundV3).map(([chainId, tokenData]) => [
+            chainId,
+            Object.fromEntries(
+              Object.entries(tokenData).map(([token, data]) => [
+                token,
+                convertSortableEventFieldsIntoRequiredFields(data),
+              ])
+            ),
+          ])
+        )
+      ),
+      bundleDepositsV3: convertTypedStringRecordIntoNumericRecord(
+        Object.fromEntries(
+          Object.entries(data.bundleDepositsV3).map(([chainId, tokenData]) => [
+            chainId,
+            Object.fromEntries(
+              Object.entries(tokenData).map(([token, data]) => [
+                token,
+                convertSortableEventFieldsIntoRequiredFields(data),
+              ])
+            ),
+          ])
+        )
+      ),
+      unexecutableSlowFills: convertTypedStringRecordIntoNumericRecord(
+        Object.fromEntries(
+          Object.entries(data.unexecutableSlowFills).map(([chainId, tokenData]) => [
+            chainId,
+            Object.fromEntries(
+              Object.entries(tokenData).map(([token, data]) => [
+                token,
+                convertSortableEventFieldsIntoRequiredFields(data),
+              ])
+            ),
+          ])
+        )
+      ),
+      bundleSlowFillsV3: convertTypedStringRecordIntoNumericRecord(
+        Object.fromEntries(
+          Object.entries(data.bundleSlowFillsV3).map(([chainId, tokenData]) => [
+            chainId,
+            Object.fromEntries(
+              Object.entries(tokenData).map(([token, data]) => [
+                token,
+                convertSortableEventFieldsIntoRequiredFields(data),
+              ])
+            ),
+          ])
+        )
+      ),
```

Comment:
Are there any options for refactoring this so it's not so repetitive?

Reply:
Fair call! https://github.com/across-protocol/sdk/pull/950/commits/869de26ac625aafd958b2cc32166ecd17d57f7e5